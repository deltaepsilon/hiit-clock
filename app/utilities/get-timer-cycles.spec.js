import getTimerCycles from './get-timer-cycles';
const powerlifting = require('../data/powerlifting.json');
const tabata = require('../data/tabata.json');

describe('getTimerCycles', () => {
  describe('Powerlifting 5x5', () => {
    const expectedCycleCount = 7;
    let result;

    beforeEach(() => {
      const timer = powerlifting.find(({ name }) => name == 'Powerlifting 5x5');

      result = getTimerCycles(timer);
    });

    it(`should have ${expectedCycleCount} cycles`, () => {
      expect(result.length).toEqual(expectedCycleCount);
    });

    it('should match snapshot', () => {
      expect(result).toMatchSnapshot();
    });
  });

  describe('Powerlifting 2x2', () => {
    const expectedCycleCount = 4;
    let result;

    beforeEach(() => {
      const timer = powerlifting.find(({ name }) => name == 'Powerlifting 2x2');

      result = getTimerCycles(timer);
    });

    it(`should have ${expectedCycleCount} cycles`, () => {
      expect(result.length).toEqual(expectedCycleCount);
    });
  });

  describe('Tabata 20/40', () => {
    let result;

    beforeEach(() => {
      const timer = tabata.find(({ name }) => name == 'Tabata 20/40');

      result = getTimerCycles(timer);
    });

    it('should have 20 cycles', () => {
      expect(result.length).toEqual(20);
    });

    it('should match snapshot', () => {
      expect(result).toMatchSnapshot();
    });
  });
});
