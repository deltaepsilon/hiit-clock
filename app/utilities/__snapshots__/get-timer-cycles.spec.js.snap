// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTimerCycles Most Densest should match snapshot 1`] = `
Array [
  Array [
    Object {
      "file": Object {},
      "id": "57c27413-19bf-45db-bc99-a27678ec8cfc",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "088ece7b-b7c4-4208-8760-43accd22f037",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "53f6adfa-2905-423e-a26a-300cd764a2e9",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "de704acb-ae53-480c-a221-16c2b6745f6a",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "15077199-bc57-45c4-8367-0d64cb81eff1",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "4b5171a5-1658-4531-9dc0-ad50487f6a8e",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "f51efbd6-0703-48ca-ab2f-518f0e41e09e",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "534589d6-e425-4d5c-8984-b5c207502373",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "5b41106b-5b1d-470f-9f88-82e0a4055616",
      "totalSeconds": 300,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "461178c2-170e-4be0-acfd-2da29f012788",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "5aa4e3fe-9768-427d-a6e3-d5cac43126ca",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "fe9d5cd9-1848-4120-86ef-e2c89e024dcb",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "3a9869ca-3b63-48d2-9526-a6826df72c72",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "87d52174-f83c-4828-931d-006e3d1abee8",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "73d2cd5e-2b58-4525-97f4-750cfed22798",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "c1efc11b-0982-4ec3-84de-c2f1a9ee4509",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "a04d6836-29b7-447a-aa35-fa4cddd71a28",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "ec0c8ced-40ed-4da4-b31a-712f4be408ef",
      "totalSeconds": 300,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "b90abc97-53dd-40fe-a1da-8282cb491e23",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "d87dd317-902b-433a-9ba6-0fe76f0fda6b",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "669926b6-9967-491e-b1f4-43e06e3ac340",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "2eaa1163-49d1-485a-ab75-d7104b83aa67",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "fc166003-4efe-4d86-af31-fd7e00db6d96",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "9694e54a-10c2-49d8-b739-471930a079f5",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "2ea21eca-27d8-450f-b641-aaa812f705ed",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "495ac7ae-cb85-4b99-b283-276b8db3ff86",
      "name": "work",
      "totalSeconds": 5,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "6796cc2d-945c-49e8-9aff-7c3fdf55b6da",
      "totalSeconds": 300,
      "type": "rest",
    },
  ],
]
`;

exports[`getTimerCycles Powerlifting 5x5 should match snapshot 1`] = `
Array [
  Array [
    Object {
      "name": "First Warmup",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Second Warmup",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Prepare for work sets",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work One",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest One",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Two",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Two",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Three",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Three",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Four",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Four",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Five",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Five",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
]
`;

exports[`getTimerCycles Powerlifting 5x5 without ending rest period should match snapshot 1`] = `
Array [
  Array [
    Object {
      "name": "First Warmup",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Second Warmup",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Prepare for work sets",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work One",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest One",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Two",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Two",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Three",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Three",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Work Four",
      "totalSeconds": 60,
      "type": "work",
    },
    Object {
      "name": "Rest Four",
      "totalSeconds": 180,
      "type": "rest",
    },
  ],
]
`;

exports[`getTimerCycles Side Splits should match snapshot 1`] = `
Array [
  Array [
    Object {
      "file": Object {},
      "id": "dd7d2b60-19aa-44fb-8d94-704c2b45b312",
      "name": "Standing forward bend",
      "totalSeconds": 30,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "ea98d94b-c658-4ad5-a015-34036aeb7942",
      "totalSeconds": 15,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "2f349a8b-ae92-46e0-b66d-296a1b0d7917",
      "name": "Left-side Sprinter",
      "totalSeconds": 120,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "386bed23-7eae-4620-a75d-4ea8d5e8d920",
      "totalSeconds": 15,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "4c333391-f8b9-4c99-92f0-4cb3c7eb07f6",
      "name": "Right-side Sprinter",
      "totalSeconds": 120,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "5302c557-ea43-4f69-bb09-05e60e84a4e8",
      "totalSeconds": 15,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "b46301fc-a852-473a-8700-ece91a46b2d4",
      "name": "Pike",
      "totalSeconds": 120,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "bcb29269-8ce6-4ec8-92b7-17846c9c3be2",
      "totalSeconds": 15,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "d2e45658-cd12-4010-8efc-5f68bc7311ce",
      "name": "Left-side Hamstring",
      "totalSeconds": 30,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "da6273c7-0769-48c4-9a52-15ee2b405983",
      "totalSeconds": 15,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "b8b2e4a3-add5-4101-90ff-de3ec8bdac0a",
      "name": "Right-side Hamstring",
      "totalSeconds": 30,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "dcaa0f18-104f-42e6-891d-3882cd93c2bc",
      "totalSeconds": 15,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "7707f3f3-5ff4-4a20-9edf-b1a5307664d6",
      "name": "Right-side Elevated Side Split",
      "totalSeconds": 120,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "a972fed9-5937-467a-a126-502ad9a1bc73",
      "totalSeconds": 45,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "5869bdf0-9733-4f78-8b80-32835965e425",
      "name": "Left-side Elevated Side Split",
      "totalSeconds": 120,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "28c8bb30-d658-452a-a3a3-1885f0266561",
      "totalSeconds": 45,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "cd65483c-c7db-42b2-81ab-092062672fa4",
      "name": "Left-side Floor Split",
      "totalSeconds": 180,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "55e5d256-4c47-4554-b276-cb2750f9df39",
      "totalSeconds": 60,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "file": Object {},
      "id": "00d69379-e3fe-47fb-b71e-2a3b915d8495",
      "name": "Right-side Floor Split",
      "totalSeconds": 180,
      "type": "work",
    },
    Object {
      "file": Object {},
      "id": "8ba2719d-f918-4cb2-be5d-1ed13f465c78",
      "totalSeconds": 60,
      "type": "rest",
    },
  ],
]
`;

exports[`getTimerCycles Tabata 20/40 should match snapshot 1`] = `
Array [
  Array [
    Object {
      "name": "Round 1",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 2",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 3",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 4",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 5",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 6",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 7",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 8",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 9",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 40",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 11",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 12",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 13",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 14",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 15",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 16",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 17",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 18",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 19",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
  Array [
    Object {
      "name": "Round 20",
      "totalSeconds": 20,
      "type": "work",
    },
    Object {
      "name": "Rest",
      "totalSeconds": 40,
      "type": "rest",
    },
  ],
]
`;
